frontend/src/components/HeaderComponent.js

```diff
@@ -1,70 +1,87 @@
-            setIsNavOpen(!isNavOpen)
-    }    
-        const history = useHistory();
-        return(
-            <React.Fragment>            
-                <div className="row topOfNav">
-                    <div className="col-12 writeup">
-                        <h3>Discover the events you are interested in...</h3>
-                    </div>
-                </div>            
-                <Navbar color="dark" dark expand="md">
-                <div className="container navHeight">
-                    <NavbarBrand href="/">EventUp</NavbarBrand>
-                    <NavbarToggler onClick={toggleNav}  />
-                    <Collapse isOpen={isNavOpen} navbar>
-                        <Nav className="ml-auto" navbar>
-                            <NavItem>
-                                <NavLink className="nav-link" to="/users/events/create">
-                                    <span> Create Event</span>
-                                </NavLink>
-                            </NavItem>
-                            {isAuth() ? 
-                               <React.Fragment>
-                               <NavItem>
-                                    <NavLink className="nav-link" to="/users/dashboard">
-                                        <span> Dashboard</span>
-                                    </NavLink>
-                                </NavItem>
-                                <Button
-                                    onClick={() => {
-                                        signout(() => {
-                                            history.push("/");
-                                        });
-                                    }}            
-                                >
-                                    Sign Out
-                                </Button>
-                                </React.Fragment>
-                                :
-                                <>
-                                <NavItem>
-                                    <NavLink className="nav-link" to="/users/login">
-                                        <span> Log in</span>
-                                    </NavLink>
-                                </NavItem>
-                                <NavItem>
-                                    <NavLink className="nav-link" to="/users/register">
-                                        <span> Sign up</span>
-                                    </NavLink>
-                                </NavItem>
-                                </>
-                            }                            
-                        </Nav>
-                    </Collapse>
-                </div>
-                </Navbar>
-            </React.Fragment>
-        )    
-}
-export default Header
\ No newline at end of file
+const Header = () => {
+  const [isNavOpen, setIsNavOpen] = useState(false);
+  const [isLoggedOut, setisLoggedOut] = useState(false);
```

Comment:
Move this state to Routes.js file and pass it here as a prop

Reply:
You can even use redux if you comfortable with it for maintaining isLoggedOut as a global state